### PR TITLE
Fix chemut memory limit and data type

### PR DESCRIPTION
**Pull request by tristanbrown**
@jkokosar @dblenkus @tjanez 
Is 32GB enough?

### CODE REVIEW

File: docs/CHANGELOG.rst [hunk 1]
```diff
@@ -58,6 +58,8 @@ Changed
 Fixed
 -----
 * Fix ``consensus_subreads`` input option in Subread process
+* Fix memory limit in vc_chemut process
```
Comment (mstajdohar) [inline docs/CHANGELOG.rst]: Please write more informative log entries. We have started to use the change log for reporting new features to the end users. Not directly of course, but I am certain @jinlajf and @rafecooks would appreciate a more verbose list of changes. I would suggest to write something like "The chemical mutagenesis analysis crashed with the out-of-box 16 GB limit for some samples <is there a common denominator for the samples where this crashed, like for Human samples or for samples larger then...>. Therefore we have increased the memory limit to 32 GB".

File: resolwe_bio/processes/variant_calling/vc_chemut.yml [hunk 2]
```diff
@@ -6,6 +6,8 @@
 - slug: vc-chemut
   name: Variant calling (CheMut)
   requirements:
+    resources:
+      memory: 32768
```
Comment (mstajdohar) [inline resolwe_bio/processes/variant_calling/vc_chemut.yml]: Are you sure we need 32 GB? Please test locally how much RAM is really needed (if you can) and how it scale with the sample size or some other dependent variable. If you can't test it locally, ask the infrastructure team to set a memory limit for this process on the server directly. We have the capability to override memory setting. I would try 20 GB next. Maybe that's enough. It's really important to keep this requirement to the minimum as it limits us to scale the number of concurrent jobs. We usually have more CPU then RAM on the worker nodes. Also, it's worth considering if we can reduce the amount of memory in use. Try to identify the command (or script) that uses all this memory and if you can either improve the script or set the input parameters.

Finally, if you have good arguments for extending the memory limit to 32 GB do so and write them in change log and the commit message (you can write more text below the first line in the commit message).

**Comment by mstajdohar** [comment]
OK, try solving this issue with the `-Xmx16g` flag and set the process' memory limit to 16 GB explicitly.

**Comment by jkokosar** [comment]
@mstajdohar please review again

File: resolwe_bio/processes/variant_calling/vc_chemut.yml [hunk 3]
```diff
@@ -174,6 +176,7 @@
 
         echo "sorting, marking duplicates, indexing"
         java \
+          -Xmx16g \
```
Comment (dblenkus) [inline resolwe_bio/processes/variant_calling/vc_chemut.yml]: Use the memory limit specified in resources.

Comment (tristanbrown) [reply resolwe_bio/processes/variant_calling/vc_chemut.yml]: Yes, isn't that what I've done here?

Comment (mstajdohar) [reply resolwe_bio/processes/variant_calling/vc_chemut.yml]: @dblenkus, please elaborate.

**Comment by mstajdohar** [comment]
@tristanbrown Were you able to test the process with the `-Xmx16g` flag? Does it work?

**Comment by tristanbrown** [comment]
@mstajdohar Yes, the -Xmx16g flag works. Regarding the suggestion from @dblenkus , I'm learning some bash and testing how to construct a -Xmx#g flag from the # given by requirements.resources.memory.

**Comment by dblenkus** [comment]
I think it would be easier to do it with Jinja. Something like: `-Xmx{{ (requirements.resources.memory / 1024) | round(method='floor') | int }}g`. Note that brackets around division are needed to get the right order of operations.

    In [1]: from jinja2 import Template

    In [2]: Template("""-Xmx{{ (16384 / 1024) | round(method='floor') | int }}g""").render()
    Out[2]: '-Xmx16g'

http://jinja.pocoo.org/docs/dev/templates/#round

**Comment by mstajdohar** [comment]
Is the `round` needed if you pipe to `int` anyway?

**Comment by dblenkus** [comment]
Yes, because it will use `common` method otherwise instead of `floor`.

**Comment by tristanbrown** [comment]
@dblenkus I was able to get the same functionality using `{{x // y}}`. I've updated the pull request.

**Comment by mstajdohar** [comment]
Quite clever, good job!

**Pull request merged**